airbyte-integrations/connectors/source-instagram/source_instagram/streams.py

```diff
@@ -165,6 +166,7 @@ class UserInsights(InstagramIncrementalStream):
     def __init__(self, **kwargs):
         super().__init__(**kwargs)
         self._end_date = pendulum.now()
+        self.time_to_exit_gracefully = False
```

Comment:
I chose boolean over throwing an exception because in the latter case I would have to catch that exception in the overridden source method `_read_incremental`/`_read_full_refresh`/etc with tons of code and new logic which would affect readability badly.
Variable renamed, code comment added.